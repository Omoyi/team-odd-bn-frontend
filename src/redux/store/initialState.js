
const state = {
  search: {
    key: '',
    payload: '',
  },
  popUpsDisplay: {
    deleteComment: 'none',
    editTrip: 'none',
    currentPopUp: 'editTrip',
  },
  pagination: {
    pageNo: 0,
    itemsPerPage: 5,
  },
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
  auth: {
    signup: {
      spinnerStatus: false,
      signupInputData: {},
      signupData: {},
      signupError: {},
    },
    signin: {
      spinnerStatus: false,
      signinInputData: {},
      signinData: {},
      signinError: {},
    },
  },
  allUsers: {
    paginatedRequest: [],
    paginationStart: 0,
    paginationEnd: 5,
  },
  trips: {
    tripRequests: {
      getCity: [],
      onewayData: {},
      onewayError: {},
      onewayInput: {},
    },
    availRequests: {
      approveRequestMessage: {},
    },
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
      paginatedRequest: [],
      paginationStart: 0,
      paginationEnd: 5,
    },
    getStats: {
      getStatsInput: {
        tripType: 1,
      },
      getStatsData: {},
      getStatsError: {},
      tripStatsCounter: {},
    },
  },
  viewProfile: {
    profile: {},
  },
  requestsData: {
    data: '',
  },
};
export default state;
