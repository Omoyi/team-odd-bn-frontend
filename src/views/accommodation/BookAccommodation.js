import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../assets/css/table.scss';
import { Redirect } from 'react-router-dom';
import {
  editRequestAction,
  paginationAction,
  editTripsAction,
} from '../../redux/actions/tripsActions/fetchRequests';
import popUpAction from '../../redux/actions/popUpAction';
import { getTripAction } from '../../redux/actions/tripsActions/onewayActions';
import { changePageNo } from '../../redux/actions/PaginationAction';
import { viewActionAccommodation } from '../../redux/actions/allAccommodation';
import { bookAccommodationAction, updateBookAccommodationInput } from '../../redux/actions/bookings/bookAnAccommodation';
import bookingInputFields from '../../constants/bookings';

class BookAccommodation extends Component {
  constructor() {
    super();
    this.inputF = (inputield, label) => (
      <div key={label} className="inputFieldsContainer">
        <div className="labelSide">{label}</div>
        <div className="inputSide">
          {inputield}
        </div>
      </div>
    );
  }

  async componentDidMount() {
    const { viewActionAccommodation, getTripAction } = this.props;
    await getTripAction();
    await viewActionAccommodation();
  }

  render() {
    const handleBooking = async () => {
      this.props.popUpAction({
        currentPopUp: 'bookAccommodation',
        bookAccommodation: 'none',
      });
      await this.props.bookAccommodationAction(this.props.bookAccommodation.bookAccommodationInput);
    };
    const {
      accomodations,
      bookAccommodation,
      updateBookAccommodationInput,
      popUpAction,
    } = this.props;
    const {
      bookAccommodationInput,
      bookAccommodationData,
    } = bookAccommodation;
    const {
      accommodationId, roomType, roomId,
    } = bookAccommodationInput;
    const rooms = accomodations
      && [].concat(...accomodations.map((request) => request.accommodationRooms));
    const curentAccommodationRoomTypes = rooms.filter(
      (room) => room.accommodationId === accommodationId,
    );
    const allAvaillableRoomTypes = curentAccommodationRoomTypes.map(
      (room) => room.roomType.toLowerCase(),
    );
    const availlableRoomTypes = allAvaillableRoomTypes.filter(
      (type, index) => allAvaillableRoomTypes.indexOf(type) === index,
    );
    const availlableRooms = rooms.filter(
      (room) => room.roomType.toLowerCase()
       === roomType
        && room.accommodationId === accommodationId,
    );
    const bookingInputList = bookingInputFields(
      availlableRoomTypes,
      accommodationId,
      availlableRooms,
      roomId,
      updateBookAccommodationInput,
      roomType,
    );
    return (
      bookAccommodationData ? <Redirect to="/bookings" /> : (
        <div className="bookingAccommodationPopupContainer">
          <div className="saveEditedRequests">
            <button
              type="button"
              id="cancel"
              onClick={() => {
                popUpAction({
                  currentPopUp: 'bookAccommodation',
                  bookAccommodation: 'none',
                });
              }}
            >
            &times;
            </button>
          </div>
          <div className="bookingHeaderContainer">
            <div>Accommodation Booking</div>
          </div>
          {bookingInputList.map((item) => this.inputF(item.inputField, item.label))}
          <div className="saveEditedRequests">
            <button
              type="button"
              id="save"
              onClick={handleBooking}
            >
           Book now!
            </button>
          </div>
        </div>
      )
    );
  }
}

export const mapStateToProps = (state) => ({
  cities: state.trips.tripRequests.getCity,
  accomodations: state.allAccomodation.accommodations,
  bookAccommodation: state.bookings.bookAccommodation,
});
const actions = {
  paginationAction,
  popUpAction,
  editTripsAction,
  getTripAction,
  editRequestAction,
  changePageNo,
  viewActionAccommodation,
  bookAccommodationAction,
  updateBookAccommodationInput,
};

export default connect(mapStateToProps, actions)(BookAccommodation);
