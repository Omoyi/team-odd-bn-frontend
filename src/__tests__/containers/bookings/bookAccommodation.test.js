import React from 'react';
import dotenv from 'dotenv';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import BookAccommodation from '../../../views/accommodation/BookAccommodation';
import initialState from '../../../redux/store/initialState';
import {
  init,
} from '../../../__mocks__/trips/requests';
import 'jest-localstorage-mock';

dotenv.config();
const mockStore = configureStore([]);
let wrapper;
let store;

describe('Common Single Request page ', () => {
  it('it Should dispatch reject button successfully', async () => {
    store = mockStore(init);
    localStorage.setItem('token', process.env.TESTING_TOKEN);

    wrapper = mount(
      <Provider store={store}>
        <Router>
          <BookAccommodation />
        </Router>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
    const selectRoomType = wrapper.find('#selectRoomType');
    const selectRoomName = wrapper.find('#selectRoomName');
    const checkInDate = wrapper.find('#checkInDate');
    const checkOutDate = wrapper.find('#checkOutDate');
    const save = wrapper.find('#save');
    const cancel = wrapper.find('#cancel');
    selectRoomType.simulate('change');
    selectRoomName.simulate('change');
    checkOutDate.simulate('change');
    checkInDate.simulate('change');
    save.simulate('click');
    cancel.simulate('click');
    expect(jest.fn().mock.calls).toBeDefined();
  });
  it('it Should dispatch reject button successfully', async () => {
    store = mockStore({
      popUpsDisplay: {
        bookAccommodation: 'none',
        deleteComment: 'none',
        editTrip: 'none',
        currentPopUp: 'editTrip',
      },
      viewProfile: { profile: '' },
      trips: { requests: { requestsData: [] }, tripRequests: { getCity: [] } },
      allAccomodation: { accommodations: [] },
      bookings: { bookAccommodation: { bookAccommodationInput: {} } },
    });
    localStorage.setItem('token', process.env.TESTING_TOKEN);

    wrapper = mount(
      <Provider store={store}>
        <Router>
          <BookAccommodation />
        </Router>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
