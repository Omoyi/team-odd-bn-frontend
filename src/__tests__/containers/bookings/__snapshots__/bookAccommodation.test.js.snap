// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Single Request page  it Should dispatch reject button successfully 1`] = `ReactWrapper {}`;

exports[`Common Single Request page  it Should dispatch reject button successfully 2`] = `ReactWrapper {}`;
